inbm/dispatcher-agent/dispatcher/schedule/sqlite_manager.py

```diff
@@ -18,79 +17,293 @@
-            logger.debug("Table not exist. Creating the table.")
-            sql = ''' CREATE TABLE repeated_task (
-                                    request_id text,
-                                    cron_duration text,
-                                    cron_minutes text,
-                                    cron_hours text,
-                                    cron_day_month text,
-                                    cron_month text,
-                                    cron_day_week text,
-                                    manifests text
-                                ); '''
-            conn.execute(sql)  # Execute the SQL command
-
-        # Add the task
-        for manifest in task.manifests:
-
-            sql = ''' INSERT INTO repeated_task(request_id, cron_duration, cron_minutes, cron_hours, cron_day_month, cron_month, cron_day_week, manifests)
-                                  VALUES(?, ?,?,?,?,?,?,?) '''
-            logger.debug(f"Adding task: {str(task.request_id), str(task.cron_duration), str(task.cron_minutes), str(task.cron_hours), str(task.cron_day_month), str(task.cron_month), str(task.cron_day_week),str(manifest)}") # noqa
-            cur.execute(sql, (str(task.request_id), str(task.cron_duration), str(task.cron_minutes), str(task.cron_hours), str(task.cron_day_month), str(task.cron_month), str(task.cron_day_week),str(manifest))) # noqa
-            conn.commit()
-        logger.debug("Task added.")
\ No newline at end of file
+            raise DispatcherException(f"Error connecting to Dispatcher Schedule database: {e}")
+    
+    def _create_single_schedule(self, ss: SingleSchedule) -> None:
+        # Add the schedule to the single_schedule table
+        logger.debug(f"Execute -> INSERT INTO single_schedule_manifest(request_id, start_time, end_time) VALUES({ss.request_id}{ss.start_time}{ss.end_time})")
+
+        sql = ''' INSERT INTO single_schedule(request_id, start_time, end_time)
+                                VALUES(?,?,?); '''
+        start_time = None if not ss.start_time else str(ss.start_time)
+        end_time = None if not ss.end_time else str(ss.end_time)
+        try:
+            self._cursor.execute(sql, (ss.request_id, start_time, end_time))
+            schedule_id = self._cursor.lastrowid
+            if not schedule_id:
+                raise DispatcherException("No schedule id was added to the single_schedule table.")
+            
+            logger.debug(f"Added schedule with id: {str(schedule_id)}, request_id: {ss.request_id}, start_time: {str(ss.start_time)}, end_time: {str(ss.end_time)}")
+        
+            # Add the manifests to the manifest table
+            manifest_ids = self._insert_manifest_to_table(ss.manifests)
+            
+            # Add the schedule_id and manifest_id to the single_schedule_manifest table
+            self._insert_single_schedule_manifest_tables(schedule_id, manifest_ids)         
+            self._commit_transaction()
```

Comment:
**<h2>$\color{#FFA500}Improvements: Code Robustness$</h2>**
**Suggestion:** Implement explicit transaction management to ensure atomicity of the database operations. The current approach relies on auto-commit mode, which might not be suitable for complex transactions.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
            try:
                self._conn.execute('BEGIN')
                self._cursor.execute(sql, (ss.request_id, start_time, end_time))
                schedule_id = self._cursor.lastrowid
                manifest_ids = self._insert_manifest_to_table(ss.manifests)
                self._insert_single_schedule_manifest_tables(schedule_id, manifest_ids)
                self._conn.commit()
            except sqlite3.Error as e:
                self._conn.rollback()
                logger.error(f"Transaction failed: {e}")
                raise DispatcherException(f"Transaction failed: {e}")
```